programs/zstdcli.c

```diff
@@ -801,21 +816,46 @@ int main(int argCount, const char* argv[])
     /* Check if benchmark is selected */
     if (operation==zom_bench) {
 #ifndef ZSTD_NOBENCH
-        BMK_setSeparateFiles(separateFiles);
-        BMK_setBlockSize(blockSize);
-        BMK_setNbWorkers(nbWorkers);
-        BMK_setRealTime(setRealTimePrio);
-        BMK_setNbSeconds(bench_nbSeconds);
-        BMK_setLdmFlag(ldmFlag);
-        BMK_setLdmMinMatch(g_ldmMinMatch);
-        BMK_setLdmHashLog(g_ldmHashLog);
+        adv.blockSize = blockSize;
+        adv.nbWorkers = nbWorkers;
+        adv.realTime = setRealTimePrio;
+        adv.nbSeconds = bench_nbSeconds;
+        adv.ldmFlag = ldmFlag;
+        adv.ldmMinMatch = g_ldmMinMatch;
+        adv.ldmHashLog = g_ldmHashLog;
         if (g_ldmBucketSizeLog != LDM_PARAM_DEFAULT) {
-            BMK_setLdmBucketSizeLog(g_ldmBucketSizeLog);
+            adv.ldmBucketSizeLog = g_ldmBucketSizeLog;
```

Comment:
Note: I think we can remove the `LDM_PARAM_DEFAULT` logic. We can leave it for now, and I can clean it up in a later patch.